src/sinks/datadog/logs.rs

```diff
@@ -38,22 +38,20 @@ impl SinkConfig for DatadogLogsConfig {
                 .port_u16()
                 .ok_or_else(|| "A port is required for endpoints".to_string())?;
 
-            let tls = if let Some(tls) = &self.tls {
-                Some(TlsSettings::from_options(&Some(tls.clone()))?)
-            } else {
-                None
-            };
-
-            (format!("{}", host), port, tls)
+            (format!("{}", host), port, self.tls.clone())
         } else {
-            (
-                "intake.logs.datadoghq.com".to_string(),
-                10516,
-                Some(TlsSettings::default()),
-            )
+            let tls = self.tls.clone().unwrap_or({
+                let mut tls = TlsConfig::default();
+                tls.enabled = Some(true);
+                tls
+            });
```

Comment:
Maybe `TlsConfig` needs a `default_enabled` function. Either that or one day we just make TLS default to enabled across the board.

Reply:
hmm honestly, I think defaulting tls to enabled like this would be a good idea. @lukesteensen and @binarylogic thoughts?

Reply:
I tend to agree but don’t understand all of the potential problems that might cause. In general defaulting towards secure makes sense.

Reply:
https://github.com/timberio/vector/issues/2528

Reply:
Agree. There may be cases where it doesn't make sense, but we should definitely do it for things that ~always go over the internet (i.e. sinks for hosted services).